runtime/wasm/src/host_exports.rs

```diff
@@ -97,6 +99,20 @@ impl<C: Blockchain> HostExports<C> {
         }
     }
 
+    fn check_entity_type_access(&self, entity_type: &EntityType) -> Result<(), HostExportError> {
+        match self.entity_type_access.allows(entity_type) {
+            true => Ok(()),
+            false => Err(HostExportError::Deterministic(anyhow!(
```

Comment:
Could this error be added as a typed error? Seems quite specific to be matched on string so I think it could be useful to have an IsolationError subtype

Reply:
It could, but given that for now we don't need to match on it I think it's fine to just use the `Deterministic` variant.